Makefile

```
@@ -99,36 +99,70 @@ doc_R_pkg = \
 
 depends = .doc/$(1) .install/$(1) .check/$(1) .test/$(1)
 
-
 ### Rules
 
 .PHONY: all install check test document shiny \
             check_base check_models check_modules 
 
+#' all: Install all packages and generate documentation
 all: install document
 
-
-check_base: $(BASE_C) 
-check_models: $(MODELS_C) 
-
-# Install base first as Modules has a circular dependency on base,
-# and then run a check on modules
-check_modules: $(BASE_I) $(MODULES_C) 
-
-document: $(ALL_PKGS_D) .doc/base/all
+#' check_base: Run R package checks on all in base/
+#' check_models: Run R package checks on all in models/
+#' check_modules: Run R package checks on all in modules/
+#    Note: Installs base first as Modules has a circular dependency on base
+check_base: $(BASE_C)
+check_models: $(MODELS_C)
+check_modules: $(BASE_I) $(MODULES_C)
+
+#' document: Generate function documentation for packages
+.PHONY: document
+document:
+	@if [ "$(filter-out $@,$(MAKECMDGOALS))" ]; then \
+		PKG="$(filter-out $@,$(MAKECMDGOALS))"; \
+		echo "Documenting package: $$PKG"; \
+		$(call doc_R_pkg, $$PKG); \
+	else \
+		echo "Documenting all packages"; \
+		$(MAKE) $(ALL_PKGS_D) .doc/base/all; \
+	fi
+
+#' install: Install all packages
+#' check: Run R package checks on all packages
+#' test: Run unit tests on all packages
+#' shiny: Install dependencies for Shiny apps
 install: $(ALL_PKGS_I) .install/base/all
 check: $(ALL_PKGS_C) .check/base/all
 test: $(ALL_PKGS_T) .test/base/all
 shiny: $(SHINY_I)
 
-# Render the PEcAn bookdown documentation
-book:
+
+#' book: Render the PEcAn bookdown documentation
+book: 
 	cd ./book_source && make build
 
 # Make the timestamp directories if they don't exist yet
 .doc .install .check .test .shiny_depends $(call depends,base) $(call depends,models) $(call depends,modules):
 	mkdir -p $@
 
+#' help: Show this help message
+help:
+	@echo "Usage: make [target]"
+	@echo ""
+	@echo "Examples:"
+	@echo "  make all     "
+	@echo "  make document"
+	@echo "  make document modules/assim.sequential  # Generate documentation for a specific package"
+	@echo ""
+	@echo "Notes:"
+	@echo "  - Components not included: cable (models), data.mining and DART (modules)."
+	@echo "  - Standard workflow: install packages, run checks, test, and document before submitting a PR."
+	@echo "  - Before submitting a PR, please ensure that all tests pass, code is linted, and documentation is up-to-date."
+	@echo ""
+	@echo "Available targets:"
+	@grep -h "^#' " $(MAKEFILE_LIST) | cut -d' ' -f2- | sort | column -t -s ':' | sed 's/^/  /'
```

I got excited about how nice this alignment looked on Mac and then realized `column` isn't universally available; this fails inside our Docker images 😿. 
I don't think alignment is worth installing more tools for; maybe instead indent all the target lines but leave the `:` ragged?
```suggestion
	@grep -h "^#' " $(MAKEFILE_LIST) | sed "s/^#' /  /" | sort
```